### ISSUE TITLE
Changes to grunt template processing?

### ISSUE THREAD

**Issue by JBCSU** [twbs/bootstrap#15745]
Was anything done to Gruntfile.js between 3.0.2 and 3.3.2 that might change the way grunt processes template strings? I forked this repo and added some tasks to Gruntfile.js to facilitate building customized versions of Bootstrap. Basically I set a global variable based on parameters to grunt, and then use template strings to use the value of that variable in path names. But when I merged in the changes from tag v3.3.2 (we were previously at 3.0.2), grunt.template.process no longer sees my global variable.

I am well and truly stuck here. Our build process depends entirely on the template process being able to see my variable. Any help / advice would be most welcome.

a T d H v A a N n K c S e.

**Comment by kkirsche** [comment]
Gruntfile changes:
https://github.com/twbs/bootstrap/commits/master/Gruntfile.js

There have been a quite a few changes to the file, though I'm unsure which are to the template process, since Nov 6, 2013 when 3.0.2 was released

**Comment by cvrebert** [comment]
Sorry, can't really do much more than advise you to review the git history yourself. We're unable to do this sort of tangential research/debugging for one person for free.
Perhaps make a series of smaller jumps instead of going straight from 3.0.2 all the way from 3.3.2?
Good luck!

**Comment by JBCSU** [comment]
Thank you for your replies. Just for the record, I wasn't asking you to do research or debugging on my behalf. I was simply asking if anyone happens to know of any changes that would affect the way grunt template processing accesses global variables. If you had intentionally made changes to it, just knowing that would've helped me as I tried to get grunt working again.

Again for the record, it seems that template strings can now only contain references to elements of grunt's config object; references to global Javascript objects throw errors now. I'm still mystified as to why, but I was able to get my tasks working again by putting a reference to the global variable into Grunt's config.
